rest-server/core/src/main/scala/com/actionml/core/store/backends/MongoDao.scala

```diff
@@ -137,6 +157,34 @@ class MongoDao[T](val collection: MongoCollection[T])(implicit ct: ClassTag[T])
     }
   }
 
+  // compare actual and required indexes and create missing ones
+  override def createIndexes(indexesDescription: Seq[(String, Indexed)]): Future[Unit] = {
+    // get information about indexes from mongo db
+    def getActualIndexesInfo(col: MongoCollection[T]): Future[Seq[(String, Indexed)]] = {
+      col.listIndexes().map { i =>
+        val keyInfo = i.get("key").get.asDocument()
+        val iName = keyInfo.getFirstKey
+        (iName, Indexed())
+      }.toFuture
+    }
+    getActualIndexesInfo(collection)
+      .map(actualIndexesInfo => indexesDescription diff actualIndexesInfo.toList)
+      .flatMap { absentIndexes =>
+        val newIndexes = absentIndexes.map { case (iName, Indexed(iOrder, ttl)) =>
+          val options = new IndexOptions()
+          if (ttl.isFinite()) options.expireAfter(ttl.toMillis, TimeUnit.MILLISECONDS)
+          IndexModel(Document(iName -> 1), options)
+        }
+        if (newIndexes.nonEmpty) {
+          logger.info(s"Creating indexes ${newIndexes.map(i => s"${i.getKeys} - ${i.getOptions}")} for collection ${collection.namespace.getFullName}")
+          collection.createIndexes(newIndexes).toFuture
+        } else Future.successful(())
+      }.recover {
+      case e: Exception =>
+        logger.error(s"Can't create indexes for ${collection.namespace.getFullName}", e)
```

Comment:
case e: exception should be banned! This is dangerous to include everywhere when you actually mean a specific exception.

The more of these we have, the more likely a fatal exception or non-expected system level exception will be caught incorrectly. For instance an OOM should NOT be caught here.

I would expect you to correct me for this rather than the other way around :-)

Reply:
OOM is an Error (subtype of Throwable), Exception can't be OOM or something that crashes jvm. ```case e => ``` is dangerous because it catches Throwables such as OOM error

Reply:
So `case e: Exception` is NOT dangerous and is considered good style?

Reply:
@pferrel I see no problems here with that code